qiskit_nature/second_q/transformers/basis_transformer.py

```diff
@@ -0,0 +1,232 @@
+        """Returns the transposed version of itself."""
+
+
+class BasisTransformer(BaseTransformer):
+    """A transformer to map from one basis to another.
+
+    Since problems have a basis associated with them (e.g. the
+    :class:`qiskit_nature.second_q.properties.bases.ElectronicBasis` in the case of the
+    :class:`qiskit_nature.second_q.problems.ElectronicStructureProblem`), this transformer can be
+    used to map from one :attr:`initial_basis` to another :attr:`final_basis`.
+
+    For example, this is how you can create an AO-to-MO transformer for an
+    :class:`qiskit_nature.second_q.problems.ElectronicStructureProblem`:
+
+    .. code-block:: python
+
+        # assuming you have the transformation coefficients from somewhere
+        ao2mo_coeff, ao2mo_coeff_b = ...
+
+        from qiskit_nature.second_q.operators import ElectronicIntegrals
+        from qiskit_nature.second_q.properties.bases import ElectronicBasis
+        from qiskit_nature.second_q.transformers import BasisTransformer
+
+        transformer = BasisTransformer(
+            ElectronicBasis.AO,
+            ElectronicBasis.MO,
+            ElectronicIntegrals.from_raw_integrals(ao2mo_coeff, h1_b=ao2mo_coeff_b),
+        )
+
+        problem_MO = transformer.transform(problem_AO)
+
+    Attributes:
+        initial_basis: the initial basis from which to map away from.
+        final_basis: the final basis into which to map into.
+        coefficients: the coefficients which transform from the initital to the final basis.
+    """
+
+    # TODO: figure out how we can make its interface non-electronic specific
+    def __init__(
+        self,
+        initial_basis: ElectronicBasis,
+        final_basis: ElectronicBasis,
+        coefficients: PolynomialTensor | Transposable,
+    ) -> None:
+        """
+        Args:
+            initial_basis: the initial basis from which to map away from.
+            final_basis: the final basis into which to map into.
+            coefficients: the coefficients which transform from the initital to the final basis.
+        """
+        self.initial_basis = initial_basis
+        self.final_basis = final_basis
+        self.coefficients = coefficients
+
+    def invert(self) -> BasisTransformer:
+        """Invert the transformer to do the reversed transformation.
+
+        Returns:
+            A new ``BasisTransformer`` mapping from :attr:`final_basis` to :attr:`initial_basis`.
+        """
+        return BasisTransformer(
+            self.final_basis,
+            self.initial_basis,
+            self.coefficients.transpose(),
+        )
+
+    def transform(self, problem: BaseProblem) -> BaseProblem:
+        if isinstance(problem, ElectronicStructureProblem):
+            return self._transform_electronic_structure_problem(problem)
+        else:
+            raise NotImplementedError(
+                f"The problem of type, {type(problem)}, is not yet supported by this transformer."
+            )
+
```

Comment:
Looking at `invert` and `transform` would an `inverse_transform` method here too be convenient?

Reply:
Do you mean something like this?
```python
def inverse_transform(self, problem):
    inv_trafo = self.invert()
    return inv_trafo.transform(problem)
```